canonical/charmhub-upload-action
# Adds the option of not uploading the charm's OCI image

Hello,

First of all, thanks for the cool action. It's really useful!

I'm proposing this change to add the possibility of updating just the charm, but not the OCI image. This can be useful for people who work with OCI images that rarely change.

I've created a new parameter called `upload-image` to be configured in the `github/workflows/integrate.yaml` file.
If not configured, the default behavior uploads the image by default, to preserve the previous behavior.
- `upload-image:  "true"` -> uploads the image
-  `upload-image: "false"` -> Doesn't upload the image
- `upload-image: "djfsdk"` -> Setting a value different from `"true"` doesn't upload the image either
- If not defined, uploads the image by default


If you wish to just upload the charm without updating the image, this would be a sample of use:

```yaml
name: Test Charmhub Upload

on:
- push
- pull_request

jobs:
  ensure-compiled:
    runs-on: ubuntu-latest
    name: Ensure ncc compilation
    steps:
      - uses: actions/checkout@v2
      - run: npm install
      - run: ./hooks/pre-commit

  test-charmhub-upload:
    runs-on: ubuntu-latest
    name: Test charmhub upload
    steps:
      - uses: actions/checkout@v2

      - uses: ./
        with:
          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
          charm-path: test-charm/
          upload-image: "false" # NEW

      - uses: ./
        with:
          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
          bundle-path: test-bundle/
```
Hope you find it useful 😄!

⚠️ It's my first time working with Github actions, probably I've overlooked something or made some dumb mistake. Feel free to suggest any changes or anything that can be useful. I'll be happy to hear you!
